src/coreclr/vm/methodtable.cpp

```diff
@@ -4200,6 +4200,9 @@ void MethodTable::AllocateRegularStaticBox(FieldDesc* pField, Object** boxedStat
         MethodTable* pFieldMT = pField->GetFieldTypeHandleThrowing().GetMethodTable();
         bool hasFixedAddr = HasFixedAddressVTStatics();
 
+        // Activate any dependent modules if necessary
+        pFieldMT->EnsureInstanceActive();
+
         // Taking a lock since we might come here from multiple threads/places
         CrstHolder crst(GetAppDomain()->GetStaticBoxInitLock());
```

Comment:
NOTE: we can also just remove the lock completely and assign the field via InterlockedCompareExchange pattern here, but that might lead to small memory leaks as GC won't collect a redundant allocation on NonGC heap in case of a race.